docs/google-test.md

```diff
@@ -0,0 +1,276 @@
+# Google Test Demo
+
+## Introduction
+
+[Google Test](https://github.com/google/googletest) is a unit testing framework for C++ programming developed by Google. It is currently being used for unit testing in the [OpenTelemetry C++](https://github.com/open-telemetry/opentelemetry-cpp) repo, along with tools like [codecov.io](http://codecov.io/) to thoroughly test the code in the repository. A primer for using Google Test can be found [here](https://github.com/google/googletest/blob/master/googletest/docs/primer.md).
+
+## Features
+
+Google Test provides a vast set of tools to be used for unit testing. The two most important are assertions and tests.
+
+### Assertions
+
+Google Test allows a developer to write test cases that use assertions to verify code behavior. Refer to the [assertions](https://github.com/google/googletest/blob/master/googletest/docs/primer.md#assertions) section in the primer for more details.
+
+### Tests
+
+To create a test, use the `TEST()` macro to define a test function. This is similar to a normal C++ function that returns no value. Google Test assertions are used inside the test function, and the result of the test is determined by the assertions used inside the test function. If any assertion fails (can be either fatal or nonfatal), then the test fails; if all assertions pass, the test succeeds. A sample test function is outlined below.
```

Comment:
If any exception/runtime error happened in `TEST()`, the test also fails. For below statement, perhaps better as "otherwise the test succeeds`?

> if all assertions pass, the test succeeds